python-avd/pyavd/_eos_designs/structured_config/base/__init__.py

```diff
@@ -635,7 +661,66 @@ def ip_tacacs_source_interfaces(self) -> None:
+            self.structured_config.ip_tacacs_source_interfaces.extend(source_interfaces)
+
+    @structured_config_contributor
+    def tacacs_servers(self) -> None:
+        """Parse AAA tacacs server configurations and update structured config with server and source interface details."""
+        if not self.inputs.aaa_settings.tacacs:
+            return
+
+        vrfs = self.inputs.aaa_settings.tacacs.vrfs
+        tacacs_source_interface = EosCliConfigGen.IpTacacsSourceInterfacesItem()
+        for server in self.inputs.aaa_settings.tacacs.servers:
+            server_vrf, source_interface = self._get_vrf_and_source_interface(
+                vrf_input=server.vrf,
+                vrfs=vrfs,
+                set_source_interfaces=True,
+                context=f"aaa_settings.tacacs.servers[ip_address={server.host}].vrf",
+            )
+            if source_interface:
+                tacacs_source_interface._update(name=source_interface, vrf=server_vrf)
+                self.structured_config.ip_tacacs_source_interfaces.append_unique(tacacs_source_interface)
+
+            self.structured_config.tacacs_servers.hosts.append_new(host=server.host, vrf=server_vrf, key=server.key)
+
+            for group in server.groups:
+                tacacs_group = self.structured_config.aaa_server_groups.obtain(group)
+                tacacs_group.type = "tacacs+"
+                tacacs_group.servers.append_new(server=server.host, vrf=server_vrf)
+
+        self.structured_config.tacacs_servers.policy_unknown_mandatory_attribute_ignore = (
+            self.inputs.aaa_settings.tacacs.policy.ignore_unknown_mandatory_attribute
+        )
+
+    @structured_config_contributor
+    def aaa_authentication(self) -> None:
+        """Assign AAA authentication configuration from inputs to structured config."""
+        if not (aaa_authentication := self.inputs.aaa_settings.authentication):
+            return
+        self.structured_config.aaa_authentication = aaa_authentication
+
+    @structured_config_contributor
+    def aaa_authorization(self) -> None:
+        """Assign AAA authorization configuration from inputs to structured config."""
+        if not (aaa_authorization := self.inputs.aaa_settings.authorization):
+            return
+        self.structured_config.aaa_authorization = aaa_authorization
+
+    @structured_config_contributor
+    def aaa_accounting(self) -> None:
+        """Assign AAA accounting configuration from inputs to structured config."""
+        if not (aaa_accounting := self.inputs.aaa_settings.accounting):
+            return
+        self.structured_config.aaa_accounting = aaa_accounting
+
+    @structured_config_contributor
+    def aaa_root_login(self) -> None:
+        """Assign AAA root login configuration from inputs to structured config."""
+        if not (aaa_root_login := self.inputs.aaa_settings.root_login):
+            return
+        self.structured_config.aaa_root.disabled = aaa_root_login.enabled
```

Comment:
the logic here should also be updated as per `disabled` `enabled` with respect to cli_gen

Reply:
maybe it should be a `not` in front of `enabled`

Reply:
today we do:

```
 self.structured_config.aaa_root.disabled = True
```

we should set the default to be `False`

so the function should be

```suggestion
        self.structured_config.aaa_root.disabled = not self.inputs.aaa_settings.root_login.enabled
```

This should be removed to not break the default:

```
        if not (aaa_root_login := self.inputs.aaa_settings.root_login):
            return
```

Reply:
updated